src/qcodes/extensions/parameters/parameter_mixin_interdependent.py

```diff
@@ -0,0 +1,195 @@
+    """
+    Mixin to define dependencies between QCoDeS parameters.
+
+    Allows a parameter to respond dynamically when other parameters change,
+    e.g., updating its label, unit, or validator when dependent values change.
+
+    Dependencies are declared via ``dependent_on`` (a list of parameter names).
+    These referenced parameters must also use this mixin. On cache changes,
+    they notify dependents via `OnCacheChangeParameterMixin`.
+
+    A user-defined ``dependency_update_method`` is called to apply custom logic
+    when dependencies change. This callable gives full control over how the
+    parameter reacts—whether adjusting metadata or triggering re-evaluation.
+
+    Attributes:
+    -----------
+    dependency_update_method : Optional[Callable[..., Any]]
+        User-provided function called when a dependency changes. Used to
+        update metadata or internal state. Optional.
+
+    dependent_on : list[str]
+        Names of other parameters this one depends on. They must use this mixin.
+
+    Notes:
+    ------
+    - ``get()`` is called automatically on dependents after updates.
+    - Parameters listed in ``dependent_on`` must already be added to the instrument.
+
+    """
+
+    _COMPATIBLE_BASES: ClassVar[list[type[ParameterBase]]] = [
+        Parameter,
+    ]
+    _INCOMPATIBLE_BASES: ClassVar[list[type[ParameterBase]]] = []
+
+    def __init__(
+        self,
+        *args: Any,
+        dependency_update_method: Callable[[], None] | None = None,
+        dependent_on: list[str] | None = None,
+        **kwargs: Any,
+    ) -> None:
+        dependency_update_method = kwargs.pop(
+            "dependency_update_method", dependency_update_method
+        )
+        dependent_on = kwargs.pop("dependent_on", dependent_on)
+
+        super().__init__(*args, **kwargs)
+
+        self.dependency_update_method = dependency_update_method
+        self.dependent_on = [] if dependent_on is None else list(dependent_on)
+
+        self._dependent_params: list[InterdependentParameterMixin] = []
+
+        self._register_dependencies()
+
+    @property
+    def dependency_update_method(self) -> Callable[[], None] | None:
+        """
+        Get the method used to update parameter attributes based on dependencies.
+
+        Returns:
+            The currently set update method or None.
+
+        """
+        return self._update_method
+
+    @dependency_update_method.setter
+    def dependency_update_method(self, method: Callable[[], None] | None) -> None:
+        """
+        Set the dependency update method.
+
+        Args:
+            method: The method or None.
+
+        Raises:
+            TypeError: If method is not callable or None.
+
+        """
+        if method is not None and not callable(method):
+            raise TypeError("dependency_update_method must be callable or None.")
+        self._update_method = method  # type: Callable[[], None] | None
```

Comment:
```suggestion
        self._update_method: Callable[[], None] | None= method
```

And probably need an import of Callable above

I think this is the last thing causing ci to fail